packages/@react-spectrum/datepicker/src/DatePickerSegment.tsx

```diff
@@ -70,6 +70,7 @@ function EditableSegment({segment, state}: DatePickerSegmentProps) {
         minWidth: !isNumeric ? null : String(segment.maxValue).length + 'ch'
       }}
       data-testid={segment.type}>
+      <span aria-hidden="true" className={classNames(styles, 'react-spectrum-DatePicker-placeholder')}>{segment.placeholder}</span>
```

Comment:
The placeholder is in the DOM at all times to reserve space for it, since the letters are not the same width as numbers.